packages/browser/src/plugins/segmentio/batched-dispatcher.ts

```diff
@@ -88,28 +91,56 @@ export default function batch(
         batch: updatedBatch,
         sentAt: new Date().toISOString(),
       }),
+    }).then((res) => {
+      if (res.status >= 500) {
+        throw new Error(`Bad response from server: ${res.status}`)
+      }
+      if (res.status === 429) {
+        const retryTimeoutStringSecs = res.headers?.get('x-ratelimit-reset')
+        const retryTimeoutMS =
+          retryTimeoutStringSecs != null
+            ? parseInt(retryTimeoutStringSecs) * 1000
```

Comment:
Not: Instead of foo != null, more idiomatic in our code base to do !foo and not use loose equality operators.